📝 Document webhooks

coyotte508 opened the pull request:
Associated PR: https://github.com/huggingface/moon-landing/pull/4477

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/hub/pr_554). All of your documentation changes will be reflected on that endpoint.

julien-c commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+# Webhooks
+
+Webhooks allow you to know when new changes happen on specific repos or repos belonging to specific users / organizations.
```
Add "inspirational" or "big picture" sentence like:

> **Webhooks are a foundation for MLOps related features. You can use this to auto-convert models, build community bots, or build CI/CD for your datasets or Spaces…**

julien-c commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+
+### Event & Domain
+
+The top-level properties `repo` and `domain` are always specified and used to determine the nature of the event.
```
```suggestion
The top-level properties `event` and `domain` are always specified and used to determine the nature of the event.
```

julien-c commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+- `"update"`
+- `"move"`
+
+`domain` specifies the domain of the event. It can be one of the following values:
+
+
+- `["repo"]` - Global events on repos: create, delete, move
+- `["repo", "content"]` - Events on the content of the repo, like new commits or tags. Triggers on new pull requests as well due to the newly created reference / commit.
+- `["repo", "config"]` - Events on the config: update space secrets, update settings, update DOIs, disabled or not, ...
+- `["community", "discussion"]` - Creating a discussion or pull request, updating the title or status
+- `["community", "discussion", "comment"]` - Creating, updating, hiding a comment
```
I would switch both ie. put domain before event

julien-c commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+- `["community", "discussion"]` - Creating a discussion or pull request, updating the title or status
+- `["community", "discussion", "comment"]` - Creating, updating, hiding a comment
+
+⚠️ **You should consider that any event on a more granular domain is an `"update"` on the encompassing domains.** ⚠️
+
+For example, any event on `["community", "discussion", "..."]` is an `"update"` on `["community", "discussion"]`.
+
+This is important in case more granularity is added to the domains by Hugging Face. For example if the domain domain `["repo", "config", "doi"]` is added, it would be accompanied by an `"add"` event when DOIs are created. 
+
+If you follow the rule above, your application will treat any event on `["repo", "config", "doi"]` as an `"update"` on `["repo", "config"]`, and will not break if Hugging Face adds more granularity to its domains.
```
is this clear for now for instance @osanseviero @nateraw? but let's ask more widely to team members)

julien-c commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+
+### Discussion
+
+The top level property `discussion` is specified on community events. Its `isPullRequest` property is a boolean indicating if the discussion is also a pull request. Sample payload:
```
```suggestion
The top level property `discussion` is specified on community events. Its `isPullRequest` property is a boolean indicating if the discussion is also a pull request (on HF, a PR is a special case of a Discussion). Sample payload:
```
+possibly link to the Hub PR & Discussion doc

SBrandeis commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+
+## Webhook payloads
+
+You will receive payloads through the `POST` method on the specified URL, in JSON format.
```
Maybe clearer this way? not sure

```suggestion
After registering a webhook, you will be notified of events via an `HTTP POST` call on the specified URL. The payload is encoded in JSON.
```

julien-c commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+- `["community", "discussion"]` - Creating a discussion or pull request, updating the title or status
+- `["community", "discussion", "comment"]` - Creating, updating, hiding a comment
+
+⚠️ **You should consider that any event on a more granular domain is an `"update"` on the encompassing domains.** ⚠️
```
or use the word `hierarchical` which kinda means this, IMO

julien-c replied:
or separate by dots and then it will be more obvious :) (and people can still filter on prefixes)

patrickvonplaten commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+
+You can configure the webhooks in your [settings](https://huggingface.co/settings/webhooks).
+
+You can watch repo changes or community events. If you watch all repos belonging to a user or an organization, you can also receive events for repo creation.
```
"receive events" => what does this mean?

coyotte508 replied:
I changed the paragraph

patrickvonplaten commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+
+`event` can take one of the following values: 
+
+- `"create"`
```
What about "duplicated" ? Guess that only exists for spaces yet no?

coyotte508 replied:
It's a `"create"` event

patrickvonplaten commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+- `"update"`
+- `"move"`
+
+`domain` specifies the domain of the event. It can be one of the following values:
```
```suggestion
`domain` can be one of the following values:
```

patrickvonplaten commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+
+For example, any event on `["community", "discussion", "..."]` is an `"update"` on `["community", "discussion"]`.
+
+This is important in case more granularity is added to the domains by Hugging Face. For example if the domain domain `["repo", "config", "doi"]` is added, it would be accompanied by an `"add"` event when DOIs are created. 
```
```suggestion
This is important in case more granularity is added to the domains by Hugging Face. For example if the domain `["repo", "config", "doi"]` is added, it would be accompanied by an `"add"` event when DOIs are created. 
```

Don't understand this either

patrickvonplaten commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+
+This is important in case more granularity is added to the domains by Hugging Face. For example if the domain domain `["repo", "config", "doi"]` is added, it would be accompanied by an `"add"` event when DOIs are created. 
+
+If you follow the rule above, your application will treat any event on `["repo", "config", "doi"]` as an `"update"` on `["repo", "config"]`, and will not break if Hugging Face adds more granularity to its domains.
```
what "rule"? "**You should consider that any event on a more granular domain is an `"update"` on the encompassing domains.** ⚠️" ? I'd maybe even link to it or write **"Rule: "** above

patrickvonplaten replied:
Ok but understand the principle now :sweat_smile: 

Think using the words "subset/superset" of would help here. Maybe: "any event on a domain will automatically trigger on event on all subsets of this domain. E.g.: ..."

coyotte508 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+# Webhooks
+
+Webhooks allow you to know when new changes happen on specific repos or repos belonging to specific users / organizations (not just your repos, but any repo!).
```
```suggestion
Webhooks allow you to react when new changes happen on specific repos or repos belonging to specific users / organizations (not just your repos, but any repo!).
```

?

patrickvonplaten commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+
+### Repo
+
+In the current version of webhooks, the top level property `repo` is always specified, as events can always be associated to a repo. Sample payload:
```
```suggestion
In the current version of webhooks, the top level property `repo` is always specified, as events can always be associated to a repo. For example, consider the following payload:
```

patrickvonplaten commented on docs/hub/webhooks.md:
````diff
@@ -0,0 +1,120 @@
+}
+```
+
+`headSha` is only sent when the domain starts with `"repo"`, it's not sent on community events.
````
Sorry what are community events? Any event on a domain that includes "community"?

julien-c replied:
it's events on discussions and PRs, ie. in the "Community" tab

maybe that "community events" is not very clear, and Discussions and PRs is more explicit. cc @gary149 too

patrickvonplaten commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+
+You can associate a secret to your webhook in two ways:
+
+- Updating the url of the webhook to add your secret as a query parameter.
```
I would have to see an example here in order to use it. Would be stuck here otherwise

patrickvonplaten reviewed (commented):
Take my feedback with a grain of salt. My front-end / networks knowledge is quite limited and I spent 0 time on googling any of the words / definitions.

sanchit-gandhi commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+You can associate a secret to your webhook in two ways:
+
+- Updating the url of the webhook to add your secret as a query parameter.
+- Setting the secret on the webhook. It will be sent as a `X-Webhook-Secret` HTTP header on every request.
```
```suggestion
- Setting the secret on the webhook. It will be sent as an `X-Webhook-Secret` HTTP header on every request.
```

sanchit-gandhi reviewed (commented):
Also not sure if I fall under your target audience! But if I do, I'd reiterate what Patrick has suggested and try and make the doc a bit more 'interactive' (either more examples or a full walkthrough)

The examples you've listed for Repo, Discussion and Comment are useful - they were the bits that were most clear for me. Until that point, it's a bit abstract and harder to understand from an ML practitioner point of view.

julien-c commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+
+## Webhook payloads
+
+After registering a webhook, you will be notified of events via an `HTTP POST` call on the specified URL. The payload is encoded in JSON.
```
yes i think it's a reasonable assumption (that the reader knows what a POST is)

That being said we could add a sentence at the top explaining what a Webhook is :

maybe something like:

> Webhooks are a way for you or your code to programmatically receive automated messages sent from the Hub when something happens.

xianbaoqian commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,120 @@
+	"content": "This adds an env key",
+	"hidden": false,
+	"url": {
+		"web": "http://huggingface.co/some-user/some-repo/discussions/4#6398872887bfcfb93a306f18"
```
api also? https://huggingface.co/docs/huggingface_hub/package_reference/hf_api#huggingface_hub.HfApi.comment_discussion

coyotte508 replied:
There is no api call to GET an individual comment at the moment

coyotte508 commented:
> Overall I think it's a good document (and agree with all the comments still open).
> 
> What I think could take this the extra mile would be to do a tutorial of a full fledge hook. Like let's say I need to update my API when the model is modified ?

And the hook could be hosted on a space!

I have something similar planned but it's not ML related (deploying a website on code changes), I think something ML-related would be better.

julien-c commented:
my thinking is that we can first do a "private beta" shipping of this (maybe using a hardcoded org like `spaces-explorers` when Spaces were in private beta) and get some team members or external fellows to experiment with this, and contribute some guides. WDYT?

coyotte508 commented:
> my thinking is that we can first do a "private beta" shipping of this (maybe using a hardcoded org like `spaces-explorers` when Spaces were in private beta) and get some team members or external fellows to experiment with this, and contribute some guides. WDYT?

not PRO users? x)

coyotte508 commented:
Thank you @patrickvonplaten @xianbaoqian @sanchit-gandhi for your suggestions!

I rewrote the docs taking into account your comments and adding a few screenshots, it's previewable at https://moon-ci-docs.huggingface.co/docs/hub/pr_554/en/webhooks. 

I also added a full payload as an example, and changed `event` and `domain` to `event.action` and `event.scope`. The `scope` is now a single string, eg `repo.config` (instead of `["repo", "config"]`)

philschmid reviewed (commented):
Looks amazing. 😍 This will allow us to build many cool features next year and examples for real MLOps automation. 

Two question from my side are:

1. Will the API be public to create webhooks? and will it then be integrated into `huggingface_hub`? 
2. Is there a way to identify the webhook request to a hub user programmatically?, e.g. `whoisthis?secret=SECRET`. => returns the userId 
-> This would allow us to add a "route" to the endpoints (UI), `/webhooks` to automatically rollout deployments.

coyotte508 commented:
>     1. Will the API be public to create webhooks? and will it then be integrated into `huggingface_hub`?

Haha cc @Wauplin 

See comment on original PR: https://github.com/huggingface/moon-landing/pull/4477#discussion_r1048361481

Opened an issue: https://github.com/huggingface/moon-landing/issues/4888. It'll be easy to make it public once it's live. 

>     2. Is there a way to identify the webhook request to a hub user programmatically?, e.g. `whoisthis?secret=SECRET`. => returns the userId
>        -> This would allow us to add a "route" to the endpoints (UI), `/webhooks` to automatically rollout deployments.

We can add `whoisthis?webhookId=XXX`, since `webhook.id` will be passed in the payload. 

Also created https://github.com/huggingface/moon-landing/issues/4889

Wauplin commented:
> Will the API be public to create webhooks? and will it then be integrated into huggingface_hub?

Thanks for the ping. Actually yes it could be even though for now it's not the priority

osanseviero commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+# Webhooks
+
+Webhooks are a foundation for MLOps related features. You can use this to auto-convert models, build community bots, or build CI/CD for your datasets or Spaces…
```
```suggestion
Webhooks are a foundation for MLOps related features. You can use this to auto-convert models, build community bots, or build CI/CD for your models, datasets, and Spaces.
```

osanseviero commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+![Settings of an individual webhook](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/webhook-settings.png)
+
+You can watch repo changes and discussions / pull requests / comments. You can even use a Space to react to webhooks!
```
```suggestion
You can watch repo changes, discussions, and Pull requests and comments. You can even use a Space to react to webhooks!
```

osanseviero commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+![image.png](https://s3.amazonaws.com/moonup/production/uploads/1671034300077-61d2f90c3c2083e1c08af22d.png)
+
+As an example, here is the full payload when a pull request is opened:
```
```suggestion
As an example, here is the full payload when a Pull request is opened:
```

osanseviero commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+`event.scope` can take one of the following values:
+
+- `"repo"` - Global events on repos. Possible values for the associated `action`: `"create"`, `"delete"`, `"update"`, `"move"`.
+- `"repo.content"` - Events on the content of the repo, like new commits or tags. Triggers on new pull requests as well due to the newly created reference / commit. The associated `action` is always `"update"`.
```
```suggestion
- `"repo.content"` - Events on the repo's content, such as new commits or tags. It triggers on new Pull requests as well due to the newly created reference/commit. The associated `action` is always `"update"`.
```

osanseviero commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+- `"repo"` - Global events on repos. Possible values for the associated `action`: `"create"`, `"delete"`, `"update"`, `"move"`.
+- `"repo.content"` - Events on the content of the repo, like new commits or tags. Triggers on new pull requests as well due to the newly created reference / commit. The associated `action` is always `"update"`.
+- `"repo.config"` - Events on the config: update space secrets, update settings, update DOIs, disabled or not, ... The associated `action` is always `"update"`.
```
```suggestion
- `"repo.config"` - Events on the config: update Space secrets, update settings, update DOIs, disabled or not, etc. The associated `action` is always `"update"`.
```

osanseviero commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+- `"repo"` - Global events on repos. Possible values for the associated `action`: `"create"`, `"delete"`, `"update"`, `"move"`.
+- `"repo.content"` - Events on the content of the repo, like new commits or tags. Triggers on new pull requests as well due to the newly created reference / commit. The associated `action` is always `"update"`.
+- `"repo.config"` - Events on the config: update space secrets, update settings, update DOIs, disabled or not, ... The associated `action` is always `"update"`.
+- `"discussion"` - Creating a discussion or pull request, updating the title or status (including merging).  Possible values for the associated `action`: `"create"`, `"delete"`, `"update"`.
```
```suggestion
- `"discussion"` - Creating a discussion or Pull request, updating the title or status, and merging.  Possible values for the associated `action`: `"create"`, `"delete"`, `"update"`.
```

osanseviero commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+- `"repo.content"` - Events on the content of the repo, like new commits or tags. Triggers on new pull requests as well due to the newly created reference / commit. The associated `action` is always `"update"`.
+- `"repo.config"` - Events on the config: update space secrets, update settings, update DOIs, disabled or not, ... The associated `action` is always `"update"`.
+- `"discussion"` - Creating a discussion or pull request, updating the title or status (including merging).  Possible values for the associated `action`: `"create"`, `"delete"`, `"update"`.
+- `"discussion.comment"` - Creating, updating, hiding a comment. Possible values for the associated `action`: `"create"`, `"update"`.
```
```suggestion
- `"discussion.comment"` - Creating, updating, and hiding a comment. Possible values for the associated `action`: `"create"`, `"update"`.
```

osanseviero commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+`event.scope` can take one of the following values:
+
+- `"repo"` - Global events on repos. Possible values for the associated `action`: `"create"`, `"delete"`, `"update"`, `"move"`.
```
I wonder if a better format for this would be a table with three columns: event.scope value, description, associated actions

osanseviero commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+`repo.headSha` is the sha of the latest commit on the repo's `main` branch. It is only sent when `event.scope` starts with `"repo"`, not on community events like discussions and comments.
+
+### Discussion
```
```suggestion
### Discussion and Pull Requests
```

osanseviero commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+### Discussion
+
+The top level property `discussion` is specified on community events. The `discussion.isPullRequest` property is a boolean indicating if the discussion is also a pull request (on HF, a PR is a special case of a Discussion). Here is an example value:
```
```suggestion
The top level property `discussion` is specified on community events (discussions and Pull requests). The `discussion.isPullRequest` property is a boolean indicating if the discussion is also a Pull request (on the Hub, a PR is a special case of a discussion). Here is an example value:
```

osanseviero commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+If you set a secret for your webhook, it will be sent along as an `X-Webhook-Secret` HTTP header on every request. Only ASCII characters are supported.
+
+<Tip>
+You can also change the URL of the webhook to add a secret to the URL. For example, setting it to `https://example.com/webhook?secret=XXX`.
```
```suggestion
You can also change the URL of the webhook to add a secret to the URL. For example, setting it to https://example.com/webhook?secret=XXX.
```

osanseviero commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+## Rate limiting
+
+Each webhook is limited to 1,000 triggers per 24 hours. You can check the  daily triggers for your webhook in your webhook settings, in the "Activity" tab.
```
Maybe add a link here for the activity tab

coyotte508 replied:
not possible - the activity tab is hf.co/settings/webhooks/xxxx/activity , it depends on the individual webhook's id

osanseviero reviewed (commented):
Nice!

julien-c reviewed (commented):
BTW did you see there's an existing `api-webhook` page

https://huggingface.co/docs/hub/api-webhook

Might make sense to redirect that old one to the new one

Slightly more concerning, that no-one ever contacted us to get access to the webhook hahaha 😹

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+# Webhooks
+
+Webhooks are a foundation for MLOps related features. You can use this to auto-convert models, build community bots, or build CI/CD for your models, datasets, and Spaces.
```
```suggestion
Webhooks are a foundation for MLOps related features. You can use them to auto-convert models, build community bots, or build CI/CD for your models, datasets, and Spaces.
```

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+Webhooks are a foundation for MLOps related features. You can use this to auto-convert models, build community bots, or build CI/CD for your models, datasets, and Spaces.
+
+They allow you to react when new changes happen on specific repos or repos belonging to specific users / organizations (not just your repos, but any repo!).
```
```suggestion
They allow you to listen for new changes on specific repos or repos belonging to particular users/organizations (not just your repos, but any repo!).
```

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+They allow you to react when new changes happen on specific repos or repos belonging to specific users / organizations (not just your repos, but any repo!).
+
+You can configure the webhooks in your [settings](https://huggingface.co/settings/webhooks):
```
```suggestion
You can create new webhooks and edit exiting ones in your webhooks [settings](https://huggingface.co/settings/webhooks):
```

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+![Settings of an individual webhook](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/webhook-settings.png)
+
+You can watch repo changes, discussions, and Pull requests and comments. You can even use a Space to react to webhooks!
```
```suggestion
Webhooks can watch for repos updates, Pull requests, discussions, and new comments. It's even possible to create a Space to react to your webhooks!
```

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+## Webhook Payloads
+
+After registering a webhook, you will be notified of events via an `HTTP POST` call on the specified URL. The payload is encoded in JSON.
```
```suggestion
After registering a webhook, you will be notified of new events via an `HTTP POST` call on the specified URL. The payload is encoded in JSON.
```

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+After registering a webhook, you will be notified of events via an `HTTP POST` call on the specified URL. The payload is encoded in JSON.
+
+You can check the last payloads sent in the activity tab of the webhook page, as well as replay webhooks:
```
```suggestion
You can view the history of payloads sent in the activity tab of the webhook settings page, it's also possible to replay past webhooks for easier debugging:
```

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+It has two sub-properties: `event.action` and `event.scope`.
+
+`event.scope` can take one of the following values:
```
```suggestion
`event.scope` will be one of the following values:
```

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+### Discussion and Pull Requests
+
+The top level property `discussion` is specified on community events (discussions and Pull requests). The `discussion.isPullRequest` property is a boolean indicating if the discussion is also a Pull request (on the Hub, a PR is a special case of a discussion). Here is an example value:
```
```suggestion
The top-level property `discussion` is specified on community events (discussions and Pull requests). The `discussion.isPullRequest` property is a boolean indicating if the discussion is also a Pull request (on the Hub, a PR is a special type of discussion). Here is an example value:
```

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+## Webhook secret
+
+Setting a webhook secret is useful to make sure payloads sent to your webhook handler come from Hugging Face.
```
```suggestion
Setting a webhook secret is useful to make sure payloads sent to your webhook handler URL are from Hugging Face.
```

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+If you set a secret for your webhook, it will be sent along as an `X-Webhook-Secret` HTTP header on every request. Only ASCII characters are supported.
+
+<Tip>
+You can also change the URL of the webhook to add a secret to the URL. For example, setting it to https://example.com/webhook?secret=XXX.
```
```suggestion
It's also possible to add the secret directly in the handler URL. For example, setting it as a query parameter: https://example.com/webhook?secret=XXX.
```

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+## Rate limiting
+
+Each webhook is limited to 1,000 triggers per 24 hours. You can check the  daily triggers for your webhook in your webhook settings, in the "Activity" tab.
```
```suggestion
Each webhook is limited to 1,000 triggers per 24 hours. You can view your usage in the Webhook settings page in the "Activity" tab.
```

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+ 
+You will see the HTTP status code and the payload of the events. You can replay events too by clicking on the `replay` button!
+
+It is possible to change the url or secret of the webhook and then replay an event; it will send the payload to the updated URL.
```
```suggestion
When changing the target URL or secret of a webhook, replaying an event will send the payload to the updated URL.
```

gary149 commented on docs/hub/webhooks.md:
```diff
@@ -0,0 +1,193 @@
+
+ ![image.png](https://s3.amazonaws.com/moonup/production/uploads/1671035382840-61d2f90c3c2083e1c08af22d.png)
+ 
+You will see the HTTP status code and the payload of the events. You can replay events too by clicking on the `replay` button!
```
```suggestion
You will see the HTTP status code and the payload of past events. You can replay those events by clicking on the `replay` button!
```

gary149 reviewed (commented):
Made a batch of copywriting suggestions.

SBrandeis reviewed (approved):
LGTM 👍

julien-c commented:
please review also the related #627
